src/DistributedWriteAheadLog/Record.cpp

```diff
@@ -13,18 +15,23 @@ namespace DB
 {
 namespace DWAL
 {
-ByteVector Record::write(const Record & record)
+ByteVector Record::write(const Record & record, const bool compressed)
 {
     ByteVector data{static_cast<size_t>((record.block.bytes() + 2) * 1.5)};
     WriteBufferFromVector wb{data};
-    NativeBlockOutputStream output(wb, 0, Block{});
+    auto may_compressed_out = compressed ? CompressedWriteBuffer(wb) : wb;
+    NativeBlockOutputStream output(may_compressed_out, 0, Block{});
 
     /// Write flags
     /// flags bits distribution
     /// [0-4] : Version
     /// [5-10] : OpCode
-    /// [11-63] : Reserved
+    /// [11-15] : Compression
```

Comment:
Why we need 5 bits for compression ?

Reply:
actually I want to encode the compression algo in the flags as well. For now, the compression algo has not been encoded.

Reply:
`flags` bits can be very precious. I don't think we shall allocate so many bits (2^5 - 1 compression algos) for compression. We probably only need support 2 - 3 compression algos.

Reply:
What's the default compression algo ? can we explicit express the algo here ?

Reply:
We shall add more unit tests in `gtest_record_serder.cpp`

Reply:
Another thing i liked to do but have no time to do is doing some microbenchmark for the record serialization and deserialization perf. The unit test is a good starting point for microbenchmark

Reply:
Yet another thing we need do before merging is "tail one message from the catalog kafka topic and dump it to a file and uncompress it manually by using the default compression algo"

Reply:
4 columns，3 rows， 100000 times of read/write
uncompressed: 17.3673 (s) compressed: 24.5077(s)